File: Antlr/src/main/scala/io/github/chaosunity/casc/parsing/expression/SuperCall.scala
package io.github.chaosunity.casc.parsing.expression

import com.google.common.collect.Lists
import io.github.chaosunity.casc.parsing.`type`.{BuiltInType, Type}
import io.github.chaosunity.casc.parsing.expression.SuperCall.SUPER_IDENTIFIER

import java.util

class SuperCall(_arguments: util.List[Expression] = Lists.newArrayList()) extends Call {
    override def arguments: util.List[Expression] = _arguments

    override def identifier: String = SUPER_IDENTIFIER

    override def `type`: Type = BuiltInType.VOID

    override def negative: Boolean = false
}

object SuperCall {
    final val SUPER_IDENTIFIER = "super"
}

File: src/main/kotlin/io/github/chaosunity/casc/parsing/node/expression/SuperCall.kt
package io.github.chaosunity.casc.parsing.node.expression

import io.github.chaosunity.casc.parsing.type.BuiltInType
import io.github.chaosunity.casc.parsing.type.Type

data class SuperCall(override val arguments: List<Argument>) : Call<SuperCall> {
    companion object {
        const val SUPER_CALL_ID = "super"
    }

    constructor() : this(listOf())

    override val identifier: String = SUPER_CALL_ID
    override val type: Type = BuiltInType.VOID
    override val negative: Boolean = false
}